plugins/logging/pkg/gateway/system.go

```diff
@@ -72,3 +78,93 @@ func (p *Plugin) UseKeyValueStore(client system.KeyValueStoreClient) {
 	p.uninstallController.Set(ctrl)
 	<-p.ctx.Done()
 }
+
+func (p *Plugin) watchGlobalCluster(client managementv1.ManagementClient) {
+	clusterClient, err := client.WatchClusters(p.ctx, &managementv1.WatchClustersRequest{})
+	if err != nil {
+		p.logger.Errorf("failed to watch clusters, exiting")
+		os.Exit(1)
+	}
+
+	p.logger.Infof("watching cluster events")
+
+	for {
```

Comment:
@kralicky  @dbason we may want "watchers" to persist their last received(/successfuly handled?) state (in the plugin KV), so that if consumers of the events are disconnected or not running, they can quickly check state differences

Reply:
I think we may want to to implement an event streaming light type pattern.  We could keep a history of events and these could be replayed on demand.  There are two options here; we could use a NATSs stream, or we could just handle this inside opni initially.

The other thing we could potentially do is have a sync initial state function which could be used on startup.  This would reset the event history and just pass out the starting state.

Reply:
Cool, we can probably omit a replay feature until we have more complex events that need something like that, my [other comment](https://github.com/rancher/opni/pull/1218#discussion_r1143485824) suggested something along the lines of an initial sync.

Reply:
Makes sense, to get this feature in for the next release I might go for the initial sync between opensearch index and the management cluster approach and we can write up an OEP if we want to start adding event replays for plugins down the line.